wrench/src/main.rs

```diff
@@ -442,211 +425,133 @@ fn main() {
+    let dim = DeviceUintSize::new(width, height);
+    wrench.update(dim);
+    thing.do_frame(&mut wrench);
 
     'outer: loop {
         if let Some(window_title) = wrench.take_title() {
             window.set_title(&window_title);
         }
 
-        if let Some(limit) = limit_seconds {
-            if (time::SteadyTime::now() - time_start) >= limit {
-                let mut block_avg_ms = block_avg_time
-                    .iter()
-                    .map(|v| as_ms(*v))
-                    .collect::<Vec<f64>>();
-                block_avg_ms.sort_by(|a, b| a.partial_cmp(b).unwrap());
-                let avg_ms =
-                    block_avg_ms.iter().fold(0., |sum, v| sum + v) / block_avg_ms.len() as f64;
-                let val_10th_pct = percentile(&block_avg_ms, 10);
-                let val_90th_pct = percentile(&block_avg_ms, 90);
-
-                println!("-    {:7} {:7} {:7}", "10th", "avg", "90th");
-                println!(
-                    "ms   {:4.3} {:4.3} {:4.3}",
-                    val_10th_pct,
-                    avg_ms,
-                    val_90th_pct
-                );
-                println!(
-                    "fps  {:4.3} {:4.3} {:4.3}",
-                    1000. / val_10th_pct,
-                    1000. / avg_ms,
-                    1000. / val_90th_pct
-                );
-                break;
-            }
-        }
-
-        let event = match window {
-            WindowWrapper::Headless(..) => glutin::Event::Awakened,
-            WindowWrapper::Window(ref window, _) => window.wait_events().next().unwrap(),
-        };
+        let mut events = Vec::new();
 
-        if let Some(limit) = limit_seconds {
-            if (time::SteadyTime::now() - time_start) >= limit {
-                break 'outer;
+        match window {
+            WindowWrapper::Headless(..) => {
+                events.push(glutin::Event::Awakened);
+            }
+            WindowWrapper::Window(ref window, _) => {
+                events.push(window.wait_events().next().unwrap());
+                events.extend(window.poll_events());
             }
         }
 
-        match event {
-            glutin::Event::Awakened => {
-                let (width, height) = window.get_inner_size_pixels();
-                let dim = DeviceUintSize::new(width, height);
-                wrench.update(dim);
-
-                let frame_num = thing.do_frame(&mut wrench);
-                unsafe {
-                    CURRENT_FRAME_NUMBER = frame_num;
-                }
+        let mut do_frame = false;
 
-                if show_help {
-                    wrench.show_onscreen_help();
+        for event in events {
+            match event {
+                glutin::Event::Closed => {
+                    break 'outer;
                 }
 
-                wrench.render();
-                window.swap_buffers();
+                glutin::Event::MouseWheel(delta, _, _) => {
```

Comment:
Since this is binary and non-positional, it seems like we should just map to +/- on the keyboard? (@kvark's suggestion)

Reply:
I couldn't get the +/- key events to work in our (old fork) of glutin. I think it's probably a bug in our old glutin fork. I could bind it to some other keys if you prefer (cursor up/down)?

Reply:
Sure that seems fine.